example_package/prog/oi.h

```diff
@@ -1,651 +1,1163 @@
+        return [&](auto main_func) {                                                           \
+            /* We need main_func in the branches to be template-dependent, hence the lambda */ \
+            if constexpr (std::is_convertible_v<decltype(main_func), int (*)()>) {             \
+                return main_func();                                                            \
+            } else {                                                                           \
+                return main_func(argc, argv);                                                  \
+            }                                                                                  \
+        }(static_cast<decltype(&only_for_type_deduction_main)>(&the_only_real_true_main));     \
+    }                                                                                          \
+    int the_only_real_true_main(__VA_ARGS__)
+
+///////////////// FORBIDDING MACROS /////////////////
+#define cin                                                         \
+    cin;                                                            \
+    static_assert(false, "Don't use cin, use oi::Scanner instead"); \
+    ::std::cin
+#define cout cout; static_assert(false, "Don't use cout, use oi::checker_verdict or oi::inwer_verdict instead or oi::bug()"); ::std::cout
+#define fstream                                                         \
+    fstream;                                                            \
+    static_assert(false, "Don't use fstream, use oi::Scanner instead"); \
+    ::std::fstream
+#define ifstream                                                         \
+    ifstream;                                                            \
+    static_assert(false, "Don't use ifstream, use oi::Scanner instead"); \
+    ::std::ifstream
+#define scanf(...) static_assert(false, "Don't use scanf(), use oi::Scanner instead")
+#define fopen(...) static_assert(false, "Don't use fopen(), use oi::Scanner instead")
+#define printf(...) static_assert(false, "Don't use printf(), use oi::checker_verdict or oi::inwer_verdict or oi::bug() instead")
+#define puts(...) \
+    static_assert(false, "Don't use puts(), use oi::checker_verdict or oi::inwer_verdict instead")
+#define exit(...) static_assert(false, "Don't use exit(), use oi::checker_verdict or oi::inwer_verdict instead")
+#define _exit(...) \
+    static_assert(false, "Don't use _exit(), use oi::checker_verdict or oi::inwer_verdict instead")
+#define _Exit(...) \
+    static_assert(false, "Don't use _Exit(), use oi::checker_verdict or oi::inwer_verdict instead")
+#define rand(...) static_assert(false, "Don't use rand(), use oi::Random")::std::rand
+#define mt19337 \
+    mt19337;    \
+    static_assert(false, "Don't use mt19337, use oi::Random")::std::mt19337
+#define mt19337_64 \
+    mt19337_64;    \
+    static_assert(false, "Don't use mt19337_64, use oi::Random")::std::mt19337_64
```

Comment:
Dorzuciłbym jeszcze `std::random_device`.
Poza tym, to możnaby go wykorzystać do seedowania, ale to tylko luźna sugestia.

Reply:
to jest zmiana do oi.h OI-owego. Proponuje narazie zostawić a jak się weźmiemy za poprawienie oi.h to wtedy dodamy.

Reply:
Nie zauważyłem, że tutaj jest po prostu wrzucony nowszy originalny oi.h.
Rzeczywiście lepiej tego u nas nie zmieniać, ale czemu tego update'a nie zupstreamujemy?